packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/ui_components/phone_number_picker.dart

```diff
@@ -0,0 +1,191 @@
+                  state: (bloc) => bloc.states.countryCode,
+                  builder: (context, countryCodeSnapshot, bloc) =>
+                      _buildPhoneNumberPrefix(
+                    context,
+                    countryCodeSnapshot.data,
+                  ),
+                ),
+                RxBlocBuilder<OnboardingPhoneBlocType, String>(
+                  state: (bloc) => bloc.states.phoneNumber,
+                  builder: (context, numberSnapshot, bloc) => Expanded(
+                      child: _buildPhoneNumberInput(
+                    context,
+                    numberSnapshot.data ?? '',
+                    phoneNumberHint,
+                  )),
+                ),
+              ],
+            ),
+            RxBlocBuilder<OnboardingPhoneBlocType, ErrorModel?>(
+              state: (bloc) => bloc.states.validationErrors,
+              builder: (context, validationErrorSnapshot, bloc) =>
+                  validationErrorSnapshot.data != null
+                      ? Padding(
+                          padding: const EdgeInsets.only(top: 16),
+                          child: Align(
+                            alignment: Alignment.centerLeft,
+                            child: Text(
+                              _translateError(
+                                  context, validationErrorSnapshot.data!),
+                              style: context.designSystem.typography.h2Reg16
+                                  .copyWith(
+                                color: context.designSystem.colors.errorColor,
+                              ),
+                            ),
+                          ),
+                        )
+                      : const SizedBox.shrink(),
+            ),
+          ],
+        ),
+      );
+
+  /// region Builders
+
+  Widget _buildPhoneNumberPrefix(
+    BuildContext context,
+    CountryCodeModel? selectedCountryCode,
+  ) =>
+      TextButton(
+        onPressed: () => _onCountryCodeSegmentPressed(
+          context,
+          selectedCountryCode,
+        ),
+        style: TextButton.styleFrom(
+          padding: const EdgeInsets.symmetric(vertical: 16),
+          shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.only(
+              topLeft: Radius.circular(_borderRadius),
+              bottomLeft: Radius.circular(_borderRadius),
+              topRight: Radius.zero,
+              bottomRight: Radius.zero,
+            ),
+          ),
+          backgroundColor: countryCodeBgColor,
+        ),
+        child: Text(
+          '+${selectedCountryCode?.code ?? countryCodeHint}',
+          style: countryCodeTextStyle ??
+              TextStyle(
+                color: context.designSystem.colors.primaryColor,
+                fontSize: 16.0,
+              ),
+        ),
+      );
+
+  Widget _buildPhoneNumberInput(
+    BuildContext context,
+    String number,
+    String hint,
+  ) =>
+      TextField(
```

Comment:
Same as above: https://medium.com/prime-holding-jsc/building-forms-in-flutter-454b8d65f48

Reply:
Done